[dinde2004] iP

dinde2004 opened the pull request:
### Thinh's chatbot
> “Success is often achieved by those who don’t know that failure is inevitable.” – Coco Chanel ([source](https://dansilvestre.com/productivity-quotes/))

Thinh's chatbot frees your mind of having to remember things to do. It's

- text-based
- easy to use
- SUPER FAST and ACCURATE response

All you need to do is:

1. download the .jar file from [here](https://github.com/dinde2004/ip/releases/tag/A-Jar).
2. execute the .jar file by using command `java -jar path/to/jar/file`.
3. add your tasks.
4. let the chatbot handle tasks for you 😉.

And the most important thing, of course, it's **FREE**!

Features:

- [x] Manage many types of tasks: todo, deadline, event.
- [x] Find the tasks with matching keyword.
- [x] And many more interesting features coming soon!

If you are a Java programmer, you can download the project from [here](https://github.com/dinde2004/ip). Here is the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

dinde2004 committed:
Add new name to chatbot. Implement greetings and exits.

dinde2004 committed:
Add the ability to echo commands to chatbot.

dinde2004 committed:
Add list

dinde2004 committed:
tasks now can mark done/not done

dinde2004 committed:
add 3 types of tasks

dinde2004 committed:
text ui testing

dinde2004 committed:
handle Exceptions

dinde2004 committed:
add delete function

dinde2004 committed:
add function to save and load data in hard disk

dinde2004 committed:
Merge branch 'branch-Level-7'

dinde2004 committed:
convert date to MMM dd yyyy format

dinde2004 committed:
Merge branch 'branch-Level-8'

dinde2004 committed:
refactor the code to make it more OOP, by adding more classes.

dinde2004 committed:
divide classes into packages

dinde2004 committed:
Merge branch 'add-gradle-support'

dinde2004 committed:
add gradle

seewhyjay commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public void run() {
+        ui.hello();
```
Might be more useful to name this method `showHelloMessage()`

seewhyjay commented on src/main/java/duke/command/Parser.java:
```diff
@@ -0,0 +1,99 @@
+            Event event = new Event(description, from, to);
+            taskList.addTask(event);
+            ui.addTask(event, taskList.size());
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Might be more intuitive to the reader if you catch the invalid command within `processCommand` rather than `addTask`

seewhyjay commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,38 @@
+        }
+    }
+
+    public int size() {
+        return this.taskList.size();
+    }
```
I'm not sure whether this is counted as a getter, but if it is, maybe `getSize` may be a more appropriate method name

seewhyjay commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    public void hello() {
+        System.out.println(HORIZONTAL_LINE);
```
Since you have a `printHorizontalLine()` method, maybe you can use that instead.

seewhyjay reviewed (commented):
Great PR! Everything is really easy to understand.

dinde2004 committed:
add JUnit tests for Deadline and Todo class

limjunxian1 commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,77 @@
+package duke.data;
+
+import duke.task.*;
```
Imported classes should always be listed explicitly even though in this case, your Storage class utilizes all the classes in your task package as it gives an excellent documentation value for the class at hand and makes the class easier to comprehend and maintain.

limjunxian1 commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,77 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Storage {
```
Write descriptive header comments for your Storage class, which I have also otherwise noticed that all your public classes lack descriptive header comments.

limjunxian1 commented on src/main/java/duke/data/Storage.java:
```diff
@@ -0,0 +1,77 @@
+        this.dataPath = dataPath;
+    }
+
+    public void writeFile(TaskList taskList) throws IOException {
```
Write descriptive header comments for your writeFile method, which I have also otherwise noticed that all your public methods lack descriptive header comments.

limjunxian1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " +
```
I like how you adhere to the coding standard that any line length should be no longer than 120 chars by practicing proper indentation, which improved readability.

limjunxian1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+package duke;
```
I like how the the ordering of import statements in all of your classes are consistent, making it easier to browse the list and determine the dependencies

limjunxian1 reviewed (commented):
Overall, I found your code easy to read because of your proper use of indentation and Egyptian braces. However, I noted a few minor coding standard violations too which I have highlighted in my comments. Otherwise, LGTM!

dinde2004 committed:
create JAR file

dinde2004 committed:
Add JavaDoc to methods and classes

dinde2004 committed:
Fix the Parser class to follow coding standard

dinde2004 committed:
Add ability to find task with a specific keyword

dinde2004 committed:
Merge branch 'branch-A-JavaDoc'

dinde2004 committed:
Merge branch 'branch-A-CodingStandard'

dinde2004 committed:
Merge branch 'branch-Level-9'

dinde2004 committed:
Add checkstyle

dinde2004 committed:
Merge branch 'branch-A-CheckStyle'

dinde2004 committed:
Improve GUI

dinde2004 committed:
Merge branch 'branch-Level-10'

dinde2004 committed:
Add some assertions to some variables

The index of the tasks must be positive integer. The adding event
command must have 2 slashes. The images of user and duke must be not
null.

Add assertions to make sure the program works properly. For example,
it is nonsense to delete the task which has index -1 or 0.

Using assertions is preferable because assertions are used to make sure
that the program works properly given the assumptions about the program
state

dinde2004 committed:
Improve the code quality

The code had some coding style error, such as a method has more than 30
line of codes, and some naming problems.

Long methods have been shortened by encapsulating each part of the
method body into other smaller methods. JavaDocs is also added to those
new methods.

Make those changes to make the code more readable.

dinde2004 committed:
Merge pull request #2 from dinde2004/branch-A-Assertions

Add some assertions to some variables

dinde2004 committed:
Merge branch 'master' of https://github.com/dinde2004/ip

dinde2004 committed:
Merge branch 'master' into branch-A-CodeQuality

dinde2004 committed:
Merge pull request #3 from dinde2004/branch-A-CodeQuality

Improve the code quality

dinde2004 committed:
Merge branch 'master' of https://github.com/dinde2004/ip

dinde2004 committed:
Implement friendly syntax for list, todo, deadline, event

dinde2004 committed:
Merge branch 'branch-C-FriendlierSyntax'

dinde2004 committed:
Add user guide and fix some minor bugs

dinde2004 committed:
Fix the README file
